### PR TITLE
Add documents to Interpretation in 20_interpret

### PR DESCRIPTION
**Pull request by edwinhung**
Add documents to Interpretation class

### CODE REVIEW

File: fastai/interpret.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# AUTOGENERATED! DO NOT EDIT! File to edit: nbs/20_interpret.ipynb (unless otherwise specified).
+# AUTOGENERATED! DO NOT EDIT! File to edit: nbs/nbs/20_interpret.ipynb (unless otherwise specified).
```
Comment (muellerzr) [inline fastai/interpret.py:1]: Run `nbdev_build_lib` from the basedir, don't specify `--fname`

File: fastai/interpret.py [hunk 2]
```diff
@@ -20,7 +20,12 @@ def plot_top_losses(x, y, *args, **kwargs):
 # Cell
 class Interpretation():
     "Interpretation base class, can be inherited for task specific Interpretation classes"
-    def __init__(self, learn, dl, losses, act=None):
+    def __init__(self,
+        learn, # Model to be interpreted
```
Comment (muellerzr) [inline fastai/interpret.py:24]:
```suggestion
        learn:Learner,
```

File: fastai/interpret.py [hunk 3]
```diff
@@ -20,7 +20,12 @@ def plot_top_losses(x, y, *args, **kwargs):
...
+        dl:DataLoader, # `DataLoader` for the model
```
Comment (muellerzr) [inline fastai/interpret.py:25]:
```suggestion
        dl:DataLoader, # `DataLoader` to run inference over
```

File: fastai/interpret.py [hunk 4]
```diff
@@ -20,7 +20,12 @@ def plot_top_losses(x, y, *args, **kwargs):
...
+        losses:TensorBase, # Losses calculated from the model
```
Comment (muellerzr) [inline fastai/interpret.py:26]:
```suggestion
        losses:TensorBase, # Losses calculated from `dl`
```

File: fastai/interpret.py [hunk 5]
```diff
@@ -34,21 +39,34 @@ def __getitem__(self, idxs):
         return inps, preds, targs, decoded, self.losses[idxs]
 
     @classmethod
-    def from_learner(cls, learn, ds_idx=1, dl=None, act=None):
+    def from_learner(cls,
+        learn, # Model used to create interpretation
+        ds_idx:int=1, # Index of `DataLoader` when dl is None
```
Comment (muellerzr) [inline fastai/interpret.py:44]:
```suggestion
        ds_idx:int=1, # Index of `learn.dls` when `dl` is None
```

File: fastai/interpret.py [hunk 6]
```diff
@@ -34,21 +39,34 @@ def __getitem__(self, idxs):
...
+    ) -> tuple:
+        "`k` largest(/smallest) losses and indexes, defaulting to all losses."
         losses, idx = self.losses.topk(ifnone(k, len(self.losses)), largest=largest)
         if items: return losses, idx, getattr(self.dl.items, 'iloc', L(self.dl.items))[idx]
         else:     return losses, idx
 
-    def plot_top_losses(self, k, largest=True, **kwargs):
-        "Show `k` largest(/smallest) preds and losses. `k` may be int, list, or `range` of desired results."
+    def plot_top_losses(self,
+        k:(int,[int]), # Number of losses to plot
```
Comment (muellerzr) [inline fastai/interpret.py:65]:
```suggestion
        k:(int,list), # Number of losses to plot
```

**Review by muellerzr — CHANGES_REQUESTED** [review]
Small nits before Benjamin takes a look

**Comment by edwinhung** [comment]
> Small nits before Benjamin takes a look

Thanks for the suggestions. I think they make sense, so I commit changes.

File: fastai/interpret.py [hunk 7]
```diff
@@ -34,21 +39,34 @@ def __getitem__(self, idxs):
         return inps, preds, targs, decoded, self.losses[idxs]
 
     @classmethod
-    def from_learner(cls, learn, ds_idx=1, dl=None, act=None):
+    def from_learner(cls,
+        learn, # Model used to create interpretation
+        ds_idx:int=1, # Index of `learn.dls` when `dl` is None
+        dl:DataLoader=None, # `Dataloader` used to make predictions
+        act=None # Activation function for prediction
```
Comment (warner-benjamin) [inline fastai/interpret.py:46]:
```suggestion
        act=None # Override default or set prediction activation function
```

File: fastai/interpret.py [hunk 8]
```diff
@@ -34,21 +39,34 @@ def __getitem__(self, idxs):
...
         _,_,losses = learn.get_preds(dl=dl, with_input=False, with_loss=True, with_decoded=False,
                                      with_preds=False, with_targs=False, act=act)
         return cls(learn, dl, losses, act)
 
-    def top_losses(self, k=None, largest=True, items=False):
-        "`k` largest(/smallest) losses and indexes, defaulting to all losses (sorted by `largest`). Optionally include items."
+    def top_losses(self,
+        k:(int,None)=None, # Number of losses and None for entire `Dataset`'s losses
+        largest:bool=True, # Sort losses by largest or smallest
+        items:bool=False # Whether to return input item for each loss
```
Comment (warner-benjamin) [inline fastai/interpret.py:57]:
```suggestion
    def top_losses(self,
        k:(int,None)=None, # Return `k` losses, defaults to all
        largest:bool=True, # Sort losses by largest or smallest
        items:bool=False # Whether to return input items
```

File: fastai/interpret.py [hunk 9]
```diff
@@ -63,7 +81,10 @@ def plot_top_losses(self, k, largest=True, **kwargs):
         #its None means that a batch knows how to show itself as a whole, so we pass x, x1
         #else: show_results(x, x1, its, ctxs=ctxs, max_n=max_n, **kwargs)
 
-    def show_results(self, idxs, **kwargs):
+    def show_results(self,
+        idxs:(TensorBase,list), # Index of predictions and targets
```
Comment (warner-benjamin) [inline fastai/interpret.py:85]: We are using list for any list-like object
```suggestion
        idxs:list, # Indices of predictions and targets
```

**Review by warner-benjamin — CHANGES_REQUESTED** [review]
A few small things to clean up. Otherwise looks good. Thanks.

**Comment by edwinhung** [comment]
> A few small things to clean up. Otherwise looks good. Thanks.

Thank you for the suggestion. It looks simpler.

**Comment by muellerzr** [comment]
Thanks!

**Pull request merged**